returnn/util/file_cache.py

```diff
@@ -397,6 +436,29 @@ class CachedFile:
     filename: str  # original filename
 
 
+@dataclass(frozen=True)
+class FileInfo:
+    """
+    Represents meta information about cached files.
+
+    We currently save the last known mtime of the source file.
+    """
+
+    mtime_ns: int
+
+    def save(self, filename: str):
+        """Saves this instance to `filename`."""
+        with open(filename, "wt") as f:
+            f.write(json.dumps(dataclasses.asdict(self)))
```

Comment:
JSON and not pickle, btw, because it is human readable.

Reply:
Note, in most other cases, we store as Python string (repr), and then use eval (or ast eval). But Json is also fine.